files/static/index.js

```diff
@@ -0,0 +1,340 @@
+"use strict";
+
+var $ = document.querySelector.bind(document);
+
+function PreloadSubmission() {
+  this.setDomainFromURLParam();
+
+  $("#domain-form").addEventListener("submit", function(ev) {
+    this.onDomainChanged();
+    ev.preventDefault();
+  }.bind(this));
+
+  $("#checkbox-owner").addEventListener("change", this.onCheckBoxChanged.bind(this));
+  $("#checkbox-subdomains").addEventListener("change", this.onCheckBoxChanged.bind(this));
+  $("#submit-form").addEventListener("submit", function(ev) {
+    this.onSubmitForm();
+    ev.preventDefault();
+  }.bind(this));
+}
+
+PreloadSubmission.prototype = {
+  // TODO: look at response codes.
+  status: function(domain) {
+    return new Promise(function(resolve, reject) {
+      var oReq = new XMLHttpRequest();
+      oReq.addEventListener("load", (function(ev) {
+        resolve(JSON.parse(oReq.response));
+      }).bind(this));
+      oReq.open("GET", "/status/" + domain);
+      oReq.send();
+    });
+  },
+
+  checkdomain: function(domain) {
+    console.log("checkdomain:" + domain);
+    return new Promise(function(resolve, reject) {
+      var oReq = new XMLHttpRequest();
+      oReq.addEventListener("load", (function(ev) {
+        resolve(JSON.parse(oReq.response));
+      }).bind(this));
+      oReq.open("GET", "/checkdomain/" + domain);
+      oReq.send();
+    });
+  },
+
+  submit: function(domain) {
+    console.log("checkdomain:" + domain);
+    return new Promise(function(resolve, reject) {
+      var oReq = new XMLHttpRequest();
+      oReq.addEventListener("load", (function(ev) {
+        resolve(JSON.parse(oReq.response));
+      }).bind(this));
+      oReq.open("GET", "/submit/" + domain);
```

Comment:
Good point about POST; I've filed a bug: https://github.com/chromium/hstspreload.appspot.com/issues/9

XSRF protection is meaningless in this case because the submission action is purely authenticated on the server; there is no user data or state.

Unless I require submissions to be tied to some sort of account (which I don't plan to do; and even IP addresses are PII), XSRF doesn't prevent anyone from making another person do something they can't do.